pkg/credentialprovider/provider.go

```diff
@@ -121,3 +121,24 @@ func (d *CachingDockerConfigProvider) Provide() DockerConfig {
 	d.expiration = time.Now().Add(d.Lifetime)
 	return d.cacheDockerConfig
 }
+
+// Reset expiration so that the provider will get configure from real provider instead of cache.
+// This is useful in node e2e testings.
+func (d *CachingDockerConfigProvider) resetExpiration() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	d.expiration = time.Now().Add(-1 * time.Hour)
```

Comment:
Minus one hour seems an arbitrary number. 
Might as well just reset `d.expiration` to its initial state: `time.Time{}`